lib/spack/spack/test/concretize.py

```diff
@@ -1953,3 +1956,106 @@ def test_external_python_extensions_have_dependency(self):
 
         assert "python" in spec["py-extension1"]
         assert spec["python"] == spec["py-extension1"]["python"]
+
+    target = spack.platforms.test.Test.default
+
+    @pytest.mark.parametrize(
+        "python_spec",
+        [
+            "python@configured",
+            "python@configured platform=test",
+            "python@configured os=debian",
```

Comment:
@scheibelp @becker33 Is it a typo or is it wanted to use `os=debian` instead of `os=debian6`? Found this while debugging a couple of unit test failures I still have in the branch I'm working on.